tests/interfaces/test_jacobian_products.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the jacobian product calculator classes.
+"""
+# pylint: disable=protected-access
+import pytest
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.interfaces.jacobian_products import JacobianProductCalculator, TransformDerivatives
+
+dev = qml.devices.experimental.DefaultQubit2()
+
+
+def inner_execute_numpy(tapes):
+    return dev.execute(tapes)
+
+
+param_shift_jpc = TransformDerivatives(inner_execute_numpy, qml.gradients.param_shift)
+hadamard_grad_jpc = TransformDerivatives(
+    inner_execute_numpy, qml.gradients.hadamard_grad, {"aux_wire": "aux"}
+)
+
+jpc_matrix = [param_shift_jpc, hadamard_grad_jpc]
+
+
+# pylint: disable=too-few-public-methods
+class TestBasics:
+    """Test initialization and repr for jacobian product calculator classes."""
+
+    def test_transform_derivatives_basics(self):
+        """Test the initialization of properties for a transform derivatives class."""
+        jpc = TransformDerivatives(
+            inner_execute_numpy, qml.gradients.hadamard_grad, {"aux_wire": "aux"}
+        )
+
+        assert isinstance(jpc, JacobianProductCalculator)
+        assert jpc._inner_execute is inner_execute_numpy
+        assert jpc._gradient_transform == qml.gradients.hadamard_grad
```

Comment:
Wouldn't we want a comparison via `is` here as well? :thinking:

Reply:
```suggestion
        assert jpc._gradient_transform is qml.gradients.hadamard_grad
```